src/decimal.rs

```diff
@@ -2127,14 +2127,14 @@ impl ToPrimitive for Decimal {
     fn to_i64(&self) -> Option<i64> {
         let d = self.trunc();
         // Quick overflow check
-        if d.hi != 0 || (d.mid & 0x8000_0000) > 0 {
```

Comment:
This is close, but isn't quite right either I don't think. I think we need to check explicitly for `i64::MIN` since that is the value which is unable to be represented.
I'll write a couple more tests and contribute to this PR.